lib/api/request/submit_survey_request.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter_survey/model/submit_survey_question_model.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'submit_survey_request.g.dart';
+
+@JsonSerializable()
+class SubmitSurveyRequest {
+  final String surveyId;
+  final List<SubmitSurveyQuestionRequest> questions;
+
+  SubmitSurveyRequest({
+    required this.surveyId,
+    required this.questions,
+  });
+
+  factory SubmitSurveyRequest.fromJson(Map<String, dynamic> json) =>
+      _$SubmitSurveyRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$SubmitSurveyRequestToJson(this);
+}
+
+@JsonSerializable()
+class SubmitSurveyQuestionRequest {
+  final String id;
+  final List<SubmitSurveyAnswerRequest> answers;
+
+  SubmitSurveyQuestionRequest({
+    required this.id,
+    required this.answers,
+  });
+
+  factory SubmitSurveyQuestionRequest.fromModel(
+      SubmitSurveyQuestionModel model) {
+    return SubmitSurveyQuestionRequest(
+      id: model.id,
+      answers: model.answers
+          .map((answer) => SubmitSurveyAnswerRequest.fromModel(answer))
+          .toList(),
+    );
+  }
+
+  factory SubmitSurveyQuestionRequest.fromJson(Map<String, dynamic> json) =>
+      _$SubmitSurveyQuestionRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$SubmitSurveyQuestionRequestToJson(this);
+}
+
+@JsonSerializable()
+class SubmitSurveyAnswerRequest {
+  final String id;
+  final String answer;
```

Comment:
What about answers that don't have `answer` (only `id`), I was expecting this would be `nullable`

Thoughts? 🙏

Reply:
I think it's ok.. if we post an answer that shouldn't be null. Though it is not clear from postman.